cli/portforward_test.go

```diff
@@ -116,106 +114,102 @@ func TestPortForward(t *testing.T) {
 		workspace = runAgent(t, client, user.UserID)
 	)
 
-	for _, c := range cases { //nolint:paralleltest // the `c := c` confuses the linter
+	for _, c := range cases {
 		c := c
 		// Delay parallel tests here because setupLocal reserves
 		// a free open port which is not guaranteed to be free
 		// between the listener closing and port-forward ready.
-		t.Run(c.name, func(t *testing.T) {
-			t.Run("OnePort", func(t *testing.T) {
-				p1 := setupTestListener(t, c.setupRemote(t))
-
-				// Create a flag that forwards from local to listener 1.
-				localAddress, localFlag := c.setupLocal(t)
-				flag := fmt.Sprintf(c.flag, localFlag, p1)
-
-				// Launch port-forward in a goroutine so we can start dialing
-				// the "local" listener.
-				inv, root := clitest.New(t, "-v", "port-forward", workspace.Name, flag)
-				clitest.SetupConfig(t, client, root)
-				pty := ptytest.New(t)
-				inv.Stdin = pty.Input()
-				inv.Stdout = pty.Output()
-				inv.Stderr = pty.Output()
-				ctx, cancel := context.WithCancel(context.Background())
-				defer cancel()
-				errC := make(chan error)
-				go func() {
-					errC <- inv.WithContext(ctx).Run()
-				}()
-				pty.ExpectMatch("Ready!")
-
-				t.Parallel() // Port is reserved, enable parallel execution.
-
-				// Open two connections simultaneously and test them out of
-				// sync.
-				d := net.Dialer{Timeout: testutil.WaitShort}
-				c1, err := d.DialContext(ctx, c.network, localAddress)
-				require.NoError(t, err, "open connection 1 to 'local' listener")
-				defer c1.Close()
-				c2, err := d.DialContext(ctx, c.network, localAddress)
-				require.NoError(t, err, "open connection 2 to 'local' listener")
-				defer c2.Close()
-				testDial(t, c2)
-				testDial(t, c1)
-
-				cancel()
-				err = <-errC
-				require.ErrorIs(t, err, context.Canceled)
-			})
+		t.Run(c.name+"_OnePort", func(t *testing.T) {
```

Comment:
This may also contribute to stability, one less test indentation. We probably need to start considering depth of tests and their interactions with each other or limiting depth to parent + subtest, not subsubtest.